src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/ProjectEncodingStringWriter.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Editor
+{
+    [Export]
+    internal class ProjectEncodingStringWriter : StringWriter
```

Comment:
This feels overkill, any reason this just wasn't a custom object non-export that took an Encoding that you passed to it? You can avoid the ExecuteSync because you are in a async method.